similarity/knn.proto

```diff
@@ -0,0 +1,14 @@
+syntax = "proto3";
```

Comment:
This should be fine for now, but in general you want to avoid needing to regenerate proto files manually. This is where build systems come into play. In general the build system will generate the current proto file at build time, completely automating this process.

----------

Also, proto definitions are language agnostic and should probably be placed in a top level directory because of this.

For example, if you're planning on this being the format that your Python job saves to datastore, the Java BE serves, and the Javascript FE requests, all of these different components will need access to the proto definition (as well as their own generated files, which again is why it's useful to have a build system automate this).

For now, I suggest leaving everything the way you have it. We'll need to consider these change if you start to actually save protos to datastore and the other components migrate off of JSON.